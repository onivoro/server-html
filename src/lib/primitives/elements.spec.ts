import {
  element
} from './element.function';
import { TElementRenderer } from '../types/element-renderer.type';

import {
  selfClosingElement
} from './self-closing-element.function';
import { TSelfClosingElementRenderer } from '../types/self-closing-element-renderer.type';

const div: TElementRenderer = element.bind(null, 'div');
const img: TSelfClosingElementRenderer = selfClosingElement.bind(null, 'img');

const logoUrl = 'https://blah.blah.com/blah.jpg';

describe('elements', () => {
  it.each([
    [div(['content goes here'], { cssClass: 'css-class-goes-here', 'data-blah': 'blahblah', style: { width: '100%', height: '30px' } })],
    [img({ height: "64", width: "auto", src: logoUrl })]
  ])('renders an HTML string', (rendering) => {
    expect(rendering).toMatchSnapshot();
  });
});
