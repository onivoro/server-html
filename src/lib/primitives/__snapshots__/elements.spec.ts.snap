// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`elements renders an HTML string 1`] = `"<div class="css-class-goes-here" data-blah="blahblah" style="width: 100%; height: 30px;">content goes here</div>"`;

exports[`elements renders an HTML string 2`] = `"<img height="64" width="auto" src="https://blah.blah.com/blah.jpg"/>"`;
